realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -15,18 +15,31 @@ constexpr auto realsense_ros_camera_version = REALSENSE_ROS_EMBEDDED_VERSION_STR
 
 PLUGINLIB_EXPORT_CLASS(realsense2_camera::RealSenseNodeFactory, nodelet::Nodelet)
 
+rs2::device global_device;
```

Comment:
No global variables. You should probably use boost to solve the problem like here: https://stackoverflow.com/questions/45137048/graceful-shutdown-on-sigint